File: scripts/fix-tourism-services.mjs
import { readFileSync, writeFileSync } from "node:fs"
import { dirname, join } from "node:path"
import { fileURLToPath } from "node:url"

const root = join(dirname(fileURLToPath(import.meta.url)), "..")
const p = join(root, "src", "data", "destinations", "tourismServices.ts")
const text = readFileSync(p, "utf8")
const lines = text.split(/(?<=\n)/)
const out = []
let seenPlanMap = false

for (const line of lines) {
  if (line.includes('id: "plan-map"')) {
    if (line.includes("ინტერაქტიული რუკა") && !seenPlanMap) {
      out.push('      { id: "plan-map", name: "ინტერაქტიული რუკა", price: 0, unit: "info" },\n')
      seenPlanMap = true
    }
    continue
  }
  out.push(line)
}

if (!seenPlanMap) {
  const fixed = []
  for (const line of out) {
    fixed.push(line)
    if (line.includes('id: "plan-weather"')) {
      fixed.push('      { id: "plan-map", name: "ინტერაქტიული რუკა", price: 0, unit: "info" },\n')
      seenPlanMap = true
    }
  }
  writeFileSync(p, fixed.join(""), "utf8")
} else {
  writeFileSync(p, out.join(""), "utf8")
}

const result = readFileSync(p, "utf8")
console.log("OK")
for (const line of result.split("\n")) {
  if (line.includes("plan-map") || line.includes("plan-weather") || line.includes("plan-offline")) {
    console.log(line)
  }
}

File: scripts/fix-tourism-services.py
from pathlib import Path

p = Path(__file__).resolve().parents[1] / "src" / "data" / "destinations" / "tourismServices.ts"
text = p.read_text(encoding="utf-8")
lines = text.splitlines(keepends=True)
out = []
seen_plan_map = False
for line in lines:
    if 'id: "plan-map"' in line:
        if "ინტერაქტიული რუკა" in line and not seen_plan_map:
            out.append(
                '      { id: "plan-map", name: "ინტერაქტიული რუკა", price: 0, unit: "info" },\n'
            )
            seen_plan_map = True
        continue
    out.append(line)

if not seen_plan_map:
    fixed = []
    for line in out:
        fixed.append(line)
        if 'id: "plan-weather"' in line:
            fixed.append(
                '      { id: "plan-map", name: "ინტერაქტიული რუკა", price: 0, unit: "info" },\n'
            )
            seen_plan_map = True
    out = fixed

p.write_text("".join(out), encoding="utf-8")
print("OK")
for line in p.read_text(encoding="utf-8").splitlines():
    if "plan-map" in line or "plan-weather" in line or "plan-offline" in line:
        print(line)
